fastapi_mail/config.py

```diff
@@ -16,20 +17,25 @@ class ConnectionConfig(Settings):
     MAIL_SSL: bool = True
     MAIL_DEBUG: conint(gt=-1, lt=2) = 0
     MAIL_FROM: EmailStr
-    MAIL_FROM_NAME: str = None
-    TEMPLATE_FOLDER: Any = None
+    MAIL_FROM_NAME: Optional[str] = None
+    TEMPLATE_FOLDER: Optional[DirectoryPath] = None
     SUPPRESS_SEND: conint(gt=-1, lt=2) = 0
     USE_CREDENTIALS: bool = True
 
-    @validator("TEMPLATE_FOLDER", pre=True)
-    def create_template_engine(cls, v):
-        template_env = None
-        if isinstance(v, str):
-            if not os.path.exists(v):
-                raise TemplateFolderDoesNotExist(
-                    f"{v} is not a valid path to an email template folder")
+    @validator("TEMPLATE_FOLDER")
+    def template_folder_validator(cls, v):
+        """Validate the template folder directory."""
+        # need to convert ``PathLike`` object
+        fp = str(v)
+        if not os.path.exists(fp) or not os.path.isdir(fp) or not os.access(fp, os.R_OK):
+            raise TemplateFolderDoesNotExist(
+                f"{v!r} is not a valid path to an email template folder")
+        return v
```

Comment:
hi, thank you for your PR. I think first we need to check TEMPLATE_FOLDER variable is None or not. If a TEMPLATE_FOLDER variable is none we don't check any validations. Otherwise, we validate the template path.

Reply:
Because the TEMPLATE_FOLDER variable is optional.

Reply:
Right you are, it has been changed